lib/Serialization/SerializedModuleLoader.cpp

```diff
@@ -1190,7 +1190,12 @@ void swift::serialization::diagnoseSerializedASTLoadFailure(
                        moduleBufferID);
     break;
   case serialization::Status::NotInOSSA:
-    // soft reject, silently ignore.
+    // Diagnose only when explicit build modules is enabled
+    if (Ctx.SerializationOpts.ExplicitModuleBuild) {
+      Ctx.Diags.diagnose(diagLoc,
```

Comment:
Why is this only diagnosed for explicit module builds?

Reply:
I have left the behavior as is for implicit module builds. It's useful for testing `-enable-ossa-modules` on a module level.

Reply:
Yes, thanks!

Reply:
Would it be possible to instead introduce a flag that allows OSSA to be lenient for testing, while making both explicit and implicit module builds strict in the same way by default?